bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,114 @@
+        response = get(complete_url)
+        data_weather = response.json()
+
+        # Now data_weather contains list of nested dictionaries, 
+        # check the value of "cod" if key is equal to 200
+        # it means the city is found otherwise, city is not found
+        if data_weather["cod"] == 200:
+            icon_id = data_weather["weather"][0]["icon"]
+            main = data_weather["main"]
+            visibility = data_weather['visibility']
+            current_temperature = main["temp"]
+            
+            kelvin_to_fahrenheit = get_kelvin_to_fahrenheit(current_temperature)
+            kelvin_to_celsius = get_kelvin_to_celsius(current_temperature)
+            
+            current_pressure = main["pressure"]
+            current_humidity = main["humidity"]
+            forcast = data_weather["weather"]
+            weather_description = forcast[0]["description"]
+            
+            embed = Embed(
+                color=self.bot.default_color,
+                title=city,
+                description=timezone_city.strftime('%m/%d/%Y %H:%M'),
+            )
+
+            embed.set_image(
+                url=f'http://openweathermap.org/img/wn/{icon_id}@2x.png'
+            )
+            embed.add_field(
+                name="Description",
+                value=capwords(weather_description),
+                inline=False
+            )
+            embed.add_field(
+                name="Visibility",
+                value=f"{visibility}m | {round(visibility*3.280839895)}ft",
+                inline=False
+            )
+            embed.add_field(
+                name="Temperature",
+                value=f"{kelvin_to_fahrenheit}°F | {kelvin_to_celsius}°C",
+                inline=False
+            )
+            embed.add_field(
+                name="Atmospheric Pressure",
+                value=f"{current_pressure} hPa",
+                inline=False
+            )
+            embed.add_field(
+                name="Humidity",
+                value=f"{current_humidity}%",
+                inline=False
+            )
+        else:
+            embed = Embed(
+                color=self.bot.default_color,
+                description=f"{city} No Found!",
+            )
+        await ctx.send(embed=embed)
+
+        
+def getTimezone(city):
+    # initialize Nominatim API
+    geolocator = Nominatim(user_agent="geoapiExercises")
+    
+    # getting Latitude and Longitude
+    location = geolocator.geocode(city)
+
+    # pass the Latitude and Longitude
+    # into a timezone_at
+    # and it return timezone
+    timezone_finder = TimezoneFinder()
+
+    result = timezone_finder.timezone_at(
+        lng=location.longitude, 
+        lat=location.latitude)
+    timezone_city = datetime.now(timezone(result))
+    return timezone_city
+
+def get_kelvin_to_celsius(kelvin):
+    kelvin_to_celsius = round((int(kelvin) - 273.15), 2)
```

Comment:
Same here, you could just return the result directly. Also, I feel like it shouldn't be the job of the function to transform the data into a int, it should be the responsibility of whatever calls this method to do it properly. 

This `it should be the responsibility of whatever calls this method to do it properly` is an interesting idea to follow. You want your function/methods to only do what they are for and nothing else.